packages/design-system/src/components/TextField/LabelMask/LabelMask.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+/**
+ * Takes the string value from an input and returns a string
+ * with appropriate date format masking applied concatenated
+ * with the date hint text, MM/DD/YYYY.
+ * @param {String} rawInput
+ * @param {Boolean} valueOnly - Defaults to false. If true,
+ * returns string without additional hint text.
+ * @returns {String}
+ */
+function maskDate(rawInput = '', valueOnly = false): any {
+  const RE_DATE = /^(\d{1,2})[\D]?(\d{1,2})?[\D]?(\d{1,4})?/;
+  const match = RE_DATE.exec(rawInput);
+  let formattedDate = '';
+  if (match) {
+    const [month, day, year] = match.slice(1);
+    formattedDate = [
+      // We treat all non-numeric characters as a delimiter. If they're using a
+      // delimiter after a month or day, we interpret that as the user supplying
+      // a single digit for month or day, which we will automatically pad for them.
+      month && month.padStart(2, '0'),
+      day && day.padStart(2, '0'),
+      year,
+    ]
+      .filter((s) => s)
+      .join('/');
+  }
+  if (valueOnly) {
+    return formattedDate;
+  }
+  const hint = 'MM/DD/YYYY';
+  const hintSub = hint.substring(formattedDate.length);
+  return formattedDate + hintSub;
+}
+
+export const BuiltInMask = {
```

Comment:
I'm thinking that this could use a better name. This was kind of a placeholder name until we came up with something better. `Mask` is already taken. Ideas:
- `MaskFunction.DATE`
- `LabelMask.Date` (if we applied them to the `LabelMask` component object itself...which devs don't actually use directly)
- `TextField.DATE_MASK` (if we applied them to the `TextField` component object itself)
- `TextField.MASK_DATE`
- `DATE_MASK` directly without an enum/object

@scheul93 and @forrestbaer, would like your input on it too

Reply:
What makes sense to me about `DATE_MASK` by itself is that unless we applied them as properties to `TextField`, developers are still going to have to import some other named thing from our package. It might as well be the constant that they want to use.